src/billing/billing-address-action-creator.ts

```diff
@@ -120,27 +121,16 @@ export default class BillingAddressActionCreator {
             marketingEmailConsent,
         }: GuestCredentials,
         options?: RequestOptions
-    ): Observable<CustomerAction> {
+    ): Observable<UpdateSubscriptionsAction> {
         if (marketingEmailConsent === undefined || marketingEmailConsent === null) {
             return empty();
         }
 
-        return concat(
-            of(createAction(CustomerActionType.UpdateCustomerRequested)),
-            defer(async () => {
-                const { body } = await this._customerRequestSender.updateCustomer({
-                    email,
-                    acceptsMarketing: marketingEmailConsent,
-                }, options);
-
-                return createAction(CustomerActionType.UpdateCustomerSucceeded, body);
-            })
-        ).pipe(
-            catchError(error => throwErrorAction(
-                CustomerActionType.UpdateCustomerFailed,
-                new UpdateCustomerError(error)
-            ))
-        );
+        return this._subscriptionActionCreator.updateSubscriptions({
+            email,
+            acceptsMarketingNewsletter: marketingEmailConsent,
+            acceptsAbandonedCartEmails: marketingEmailConsent,
```

Comment:
How come `acceptsMarketingNewsletter` and `acceptsAbandonedCartEmails` both reflect the same `marketingEmailConsent` parameter? Our UI might have a single checkbox, but I guess there's a reason for our API to distinguish between the two therefore we should reflect that on the SDK level.

Reply:
Mostly thought about setting both values because we already released this API (`marketingEmailConsent`) and this way, avoid breaking changes. The name is generic enough to encompass both newsletter and abandoned cart emails.

However, the truth is, that this was not having any effect when passed as it requires a CP setting to be turned on (hidden behind an experiment). Happy to "break" the API.

Reply:
I see. Before we have one flag, now we have two. I think what you can do is to introduce two new flags `acceptsMarketingNewsletter` and `acceptsAbandonedCartEmails` and deprecate the old flag `marketingEmailConsent`. For backward compatibility, if `marketingEmailConsent` is passed as `true`, both `acceptsMarketingNewsletter` and `acceptsAbandonedCartEmails` are passed as `true`. The caller can otherwise pass the values separately using the new flags.

I think it's worth having the new method `updateSubscriptions` because the SDK serves as a JS client for the checkout storefront API. So if there's an endpoint for it, we should have a corresponding JS method for it unless there's a good reason not to.